### PR TITLE
Write tests that cover routes and controllers

### PR DESCRIPTION
**Pull request by orangeboy55555**
The tests that were written are integration tests. They cover the Post controller and routes except for one line in the controller, which I do not know how to cover. To do so, I need to make Post.find() throw an exception in post-model.js, which is not possible since the method does not accept anything.

The CI pipeline must be changed because we need a new environmental variable called TEST_TYPE, which is either UNIT or INTEGRATION. Feel free to change the name of this variable. For example, you could call it IS_UNIT = TRUE or FALSE.

### CODE REVIEW

**Review by DaTanUmanitoba — APPROVED** [review]
it seems all good to merge.

File: backend/tests/post-route-test.js [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+            .then(() => {console.log("Success to connect mongodb");})
+            .catch(() => {console.log("Fail to connect mongodb")});
+
+        app.use(bodyParser.json({extended: true}));
+        app.use(bodyParser.urlencoded({extended: true}));
+        app.use(cors())
+            
+        app.use("/api", apiRouter)
+            
+        // just for sample test, delete before sprint2 due
```
Comment (DaTanUmanitoba) [inline backend/tests/post-route-test.js:89]: this needs to be gone

**Review by WeiyuSun — COMMENTED** [review]
Looks good for me

**Review by WeiyuSun — APPROVED** [review]
looks good for me

**Pull request merged**